solution/build.sh

```diff
@@ -0,0 +1,183 @@
+	ZOOKEEPER_CRD_FILE=${CRDS_DIR}/zookeeper_operator_crd.yaml
+	wget -O ${BUILD_ROOT}/zookeeper-operator.tar.gz https://github.com/pravega/zookeeper-operator/archive/v${ZOOKEEPER_VERSION}.tar.gz
+	mkdir ${BUILD_ROOT}/zookeeper-operator
+	tar -C ${BUILD_ROOT}/zookeeper-operator --strip-components=1 -xf ${BUILD_ROOT}/zookeeper-operator.tar.gz
+	cp ${BUILD_ROOT}/zookeeper-operator/deploy/crds/zookeeper_v1beta1_zookeepercluster_crd.yaml ${ZOOKEEPER_CRD_FILE}
+	echo --- >> ${ZOOKEEPER_CRD_FILE}
+	cat ${BUILD_ROOT}/zookeeper-operator/deploy/all_ns/rbac.yaml >> ${ZOOKEEPER_CRD_FILE}
+	echo --- >> ${ZOOKEEPER_CRD_FILE}
+	cat ${BUILD_ROOT}/zookeeper-operator/deploy/all_ns/operator.yaml >> ${ZOOKEEPER_CRD_FILE}
+}
+
+
+function gen_operator_yaml()
+{
+	# we need to escape / with \/ in our sed command
+	sed "s/REPLACE_IMAGE/zenko-operator:${OPERATOR_TAG}/" operator.yaml > ${ISO_ROOT}/operator/operator.yaml
+}
+
+function gen_operator_config_yaml()
+{
+	CONFIG_PATH=${ISO_ROOT}/operator/operator-config.yaml
+
+	echo apiVersion: solutions.metalk8s.scality.com/v1alpha1 > ${CONFIG_PATH}
+	echo kind: OperatorConfig >> ${CONFIG_PATH}
+	echo repositories: >> ${CONFIG_PATH}
+  	echo "  ${VERSION_FULL}:" >> ${CONFIG_PATH}
+    	echo "    - endpoint: metalk8s-registry/${PRODUCT_LOWERNAME}-${VERSION_FULL}" >> ${CONFIG_PATH}
+      	echo "    images:" >> ${CONFIG_PATH}
+	for dep in ${DEP_IMAGES[@]}; do
+		SHORT_DEP=${dep##*/}
+        	echo "      - ${SHORT_DEP}" >> ${CONFIG_PATH}
+	done
+}
+
+function copy_image()
+{
+	IMAGE_NAME=${1##*/}
+	FULL_PATH=${IMAGES_ROOT}/${IMAGE_NAME/:/\/}
+	mkdir -p ${FULL_PATH}
+	${SKOPEO} ${SKOPEO_OPTS} copy \
+		--format v2s2 --dest-compress \
+		--src-daemon-host ${DOCKER_SOCKET} \
+		docker-daemon:${1} \
+		dir:${FULL_PATH}
+}
+
+function dedupe()
+{
+	${HARDLINK} -c ${IMAGES_ROOT}
+}
+
+function build_registry_config()
+{
+	docker run \
+    	 	--name static-oci-registry \
+	 	--mount type=bind,source=${ISO_ROOT}/images,destination=/var/lib/images \
+		--mount type=bind,source=${ISO_ROOT},destination=/var/run \
+	 	--rm \
+    		docker.io/nicolast/static-container-registry:latest \
+		python3 /static-container-registry.py --omit-constants /var/lib/images
+	CONFIG_FILE=${ISO_ROOT}/static-container-registry.conf
+	mv ${CONFIG_FILE} ${ISO_ROOT}/registry-config.inc
+}
+
+function build_iso()
+{
+	mkisofs -output ${ISO} \
+		-quiet \
+		-rock \
+		-joliet \
+		-joliet-long \
+		-full-iso9660-filenames \
+		-volid "${PRODUCT_NAME} ${VERSION_FULL}" \
+		--iso-level 3 \
+		-gid 0 \
+		-uid 0 \
+		-input-charset iso8859-1 \
+		-output-charset iso8859-1 \
+		${ISO_ROOT}
+	sha256sum ${ISO} > ${ISO_ROOT}/SHA256SUM
+	echo ISO File at ${ISO}
+	echo SHA256 for ISO:
+	cat ${ISO_ROOT}/SHA256SUM
+}
+
+# run everything in order
+clean
+mkdirs
+gen_manifest_yaml
+copy_yamls
+download_dependency_yamls
+gen_operator_yaml
+gen_operator_config_yaml
+for img in ${DEP_IMAGES[@]}; do
```

Comment:
When iterating over an array in Bash, it's a good practice to always protect the variable with double quotes (even if there is no spaces), to avoid unwanted behavior because of white characters:
```suggestion
for img in "${DEP_IMAGES[@]}"; do
```